contents/tutorials/vue-analytics.md

```diff
@@ -0,0 +1,279 @@
+
+```js file=main.js
+import { createApp } from 'vue'
+import App from './App.vue'
+import router from './router'
+
+createApp(App).use(router).mount('#app')
+```
+
+The basic setup is now complete. Run `npm run serve` to see your app.
+
+![Basic Vue app](../images/tutorials/vue-analytics/basic-app.png)
+
+## Adding PostHog
+
+> This tutorial shows how to integrate PostHog with `Vue 3`. If you're using `Vue 2`, see [our Vue docs](/docs/libraries/vue-js) for how to integrate PostHog.
+
+With our app set up, it’s time to install and set up PostHog. If you don't have a PostHog instance, you can [sign up for free](https://app.posthog.com/signup). 
+
+First install `posthog-js` .
+
+```bash
+npm install posthog-js
+```
+
+Create a new [plugin](https://vuejs.org/guide/reusability/plugins) by creating a new folder in your base directory called `plugins` and then a new file `posthog.js`:
+
+```bash
+mkdir plugins
+cd plugins 
+touch posthog.js
+```
+
+Add the following code to your `posthog.js` file:
+
+```js file=plugins/posthog.js
+import posthog from "posthog-js";
+
+export default {
+  install(app) {
+    app.config.globalProperties.$posthog = posthog.init(
+      "<ph_project_api_key>",
+      {
+        api_host: "<ph_instance_address>",
+      }
+    );
+  },
+};
+```
+
+Replace `<ph_project_api_key>` and `<ph_instance_address>` with your your PostHog API key and host. You can find these in your [project settings](https://app.posthog.com/settings/project).
+
+Finally, activate your plugin in `main.js`:
+
+```js file=main.js
+import { createApp } from 'vue'
+import App from './App.vue'
+import router from './router'
+import posthogPlugin from '../plugins/posthog';
+
+createApp(App)
+.use(posthogPlugin)
+.use(router)
+.mount('#app');
+```
+
+Once you’ve done this, reload your app and click the buttons a few times. You should see events appearing in the [PostHog events explorer](https://app.posthog.com/events).
+
+<ProductScreenshot
+  imageLight={EventsLight} 
+  imageDark={EventsDark} 
+  alt="Events in PostHog" 
+  classes="rounded"
+/>
+
+## Capturing pageviews
+
+You might notice that moving between pages only captures a single pageview event. This is because PostHog only captures pageview events when [page load](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) is fired. Since Vue creates a single-page app, this only happens once, and the Vue router handles subsequent page changes.
+
+If we want to capture every route change, we must write code to capture pageviews that integrates with the router.
+
+In `router/index.js`, set up PostHog to capture pageviews in the `router.afterEach` function. Additionally, you can use `nextTick` so that the capture event fires only after the page is mounted.
+
+```js router/index.js
+import { createRouter, createWebHistory } from 'vue-router'
+import { nextTick } from 'vue';
+
+// rest of your code
+
+router.afterEach((to, from, failure) => {
+  if (!failure && router.app) {
```

Comment:
router.app is undefined for me here, so this isn't running

Reply:
Yikes. This happened for me too now, and not sure how I even got to it work in the first place 🤷🏻‍♂️ I've updated the code now with a new approach. Thanks for catching for this